node.go

```diff
@@ -135,53 +53,65 @@ func (n *SphinxNode) PrefixFreeDecode(s []byte) (int, []byte, []byte) {
 	return Failure, nil, nil
 }
 
-// Unwrap unwraps a layer of encryption from a sphinx packet
-// and upon success returns an UnwrappedMessage, otherwise an error.
-func (n *SphinxNode) Unwrap(packet *SphinxPacket) (*UnwrappedMessage, error) {
+// UnwrappedMessage is produced by SphinxNode's Unwrap method
+type UnwrappedMessage struct {
+	ProcessAction int
+	Alpha         []byte
+	Beta          []byte
+	Gamma         []byte
+	Delta         []byte
+	NextHop       []byte
+	ClientID      []byte
+	MessageID     []byte
+}
+
+// SphinxNodeState is an interface for sphinx mix node state
+type SphinxNodeState interface {
+	// Get returns true if the hash value is present in the map
+	Get([32]byte) bool
+	// Set sets a hash value in the map
+	Set([32]byte)
+	// Flush flushes the map
+	Flush()
+	// GetPrivateKey returns the private key
+	GetPrivateKey() [32]byte
+}
+
+func SphinxPacketUnwrap(params *SphinxParams, state SphinxNodeState, packet *SphinxPacket) (*UnwrappedMessage, error) {
```

Comment:
worth an explicit comment that it isn't thread safe if you expect clients to use it concurrently.